github/README.md

```diff
@@ -0,0 +1,90 @@
+# Github Authentication and Authorization
+
+## Deployment
+
+Syndesis the project files in git. By default we use Github which at the moment is really the only
+git provider we support. At deployment time of Syndesis the Administrator creates a "Client App" in
+the Github account that will be used by Syndesis, see also http://syndesis.io/quickstart/#github-registered-application
+for more info. At that time the GITHUB_CLIENT_ID AND GITHUB_CLIENT_SECRET
+settings are made available to the Syndesis installation. For more info on that see 
+http://syndesis.io/quickstart/#deployment-instructions
+
+## Keycloak
+
+With the information above a Keycloak provider is created which at login time issues an OAuthToken that
+will be set on a header in each request. For this token to be able to be used by git, it needs to have
+'repo' scope. This scope info is set when Keycloak interacts with Github. See the table below for a full
+list of scopes supported by Github [1] and [2].
+
+|Name|	Description|
```

Comment:
Isn't it too confusing to list here all scopes ? Wouldn't it be simpler to just say which scope we need and then refer for the full list to GitHub docs ?

Also I don't think that in the current setup Keycloak interacts with GitHub, it just holds the GitHub token and returns it to syndesis-rest, which then in turn uses it to contact GitHub. (right?)